src/formats/html.ts

```diff
@@ -75,10 +75,85 @@ export class HtmlImporter extends FormatImporter {
 			return;
 		}
 
+		const fileLookup = new Map<string, { file: PickedFile, tFile: TFile }>;
+		ctx.reportProgress(0, files.length);
 		for (let i = 0; i < files.length; i++) {
 			if (ctx.isCancelled()) return;
 			ctx.reportProgress(i, files.length);
-			await this.processFile(ctx, folder, files[i]);
+			const file = files[i];
+			const tFile = await this.processFile(ctx, folder, file);
+			if (tFile) {
+				fileLookup.set(file instanceof NodePickedFile ? nodeUrl.pathToFileURL(file.filepath).href : file.name, { file, tFile });
+			}
+		}
+
+		if (fileLookup.size > 0) {
+			const { metadataCache } = this.app;
+			// @ts-ignore
+			if (!metadataCache.computeMetadataAsync) {
+				const aFile = fileLookup.values().next().value;
+				const resolved = new Promise<void>(resolve => {
+					const ref = this.app.metadataCache.on('resolved', () => {
+						this.app.metadataCache.offref(ref);
+						resolve();
+					});
+				});
+				const appended = '\n';
+				await this.app.vault.append(aFile, appended);
+				await resolved;
+				await this.app.vault.process(aFile, data => data.endsWith(appended) ? data.slice(0, -appended.length) : data);
+			}
+
+			for (const [fileKey, { file, tFile }] of fileLookup) {
+				if (ctx.isCancelled()) return;
+				try {
+					// Attempt to parse links using MetadataCache
+					let mdContent = await this.app.vault.cachedRead(tFile);
+					let cache;
+					// @ts-ignore
+					if (metadataCache.computeMetadataAsync) {
+						// @ts-ignore
+						cache = await metadataCache.computeMetadataAsync(stringToUtf8(mdContent)) as CachedMetadata;
+					}
+					else {
+						cache = metadataCache.getFileCache(tFile);
+						if (!cache) continue;
+					}
+
+					// Gather changes to make to the document
+					const changes = [];
+					if (cache.links) {
+						for (const { link, position, displayText } of cache.links) {
+							const { path, subpath } = parseLinktext(link);
+							let linkKey;
+							if (nodeUrl) {
+								const url = new URL(encodeURI(path), fileKey);
+								url.hash = '';
+								url.search = '';
+								linkKey = decodeURIComponent(url.href);
+							}
+							else {
+								linkKey = parseFilePath(path.replace(/#/gu, '%23')).name;
+							}
+							if (fileLookup.has(linkKey)) {
+								const newLink = this.app.fileManager.generateMarkdownLink(fileLookup.get(linkKey)!.tFile, tFile.path, subpath, displayText);
```

Comment:
Please rewrite this to avoid the assertion. For example, get the value from `fileLookup` first, then check to see if it is defined instead of using `.has`.

Reply:
There is also one outside of my diff here: https://github.com/obsidianmd/obsidian-importer/blob/cfdfac1bd55eec84b2aca8cc35defae6365cc730/src/formats/html.ts#L172